oauth2-authorization-server/src/main/java/org/springframework/security/config/annotation/web/configurers/oauth2/server/authorization/OAuth2AuthorizationServerConfigurer.java

```diff
@@ -384,6 +411,18 @@ private static void validateProviderSettings(ProviderSettings providerSettings)
 		return jwtCustomizer;
 	}
 
+	private static <B extends HttpSecurityBuilder<B>> UserInfoClaimsMapper getUserInfoClaimsMapper(B builder) {
+		UserInfoClaimsMapper userInfoClaimsMapper = builder.getSharedObject(UserInfoClaimsMapper.class);
+		if (userInfoClaimsMapper == null) {
+			userInfoClaimsMapper = getOptionalBean(builder, UserInfoClaimsMapper.class);
```

Comment:
Since I suggested reducing the visibility of this class, the user cannot provide a bean of this type. I haven't tested it, but here's a suggested alternative implementation of this method using `ResolvableType` that would allow a user to provide a bean of type `Function<Object, OidcUserInfo>`:

```java
	static <B extends HttpSecurityBuilder<B>> Function<Object, OidcUserInfo> getUserInfoClaimsMapper(B builder) {
		OidcUserInfoClaimsMapper userInfoClaimsMapper = builder.getSharedObject(OidcUserInfoClaimsMapper.class);
		if (userInfoClaimsMapper == null) {
			ResolvableType type = ResolvableType.forClassWithGenerics(Function.class, Object.class, OidcUserInfo.class);
			Function<Object, OidcUserInfo> mapperFunction = getOptionalBean(builder, type);
			if (mapperFunction == null) {
				mapperFunction = (principal) -> OidcUserInfo.builder()
						.subject(principal.toString())
						.build();
			}
			userInfoClaimsMapper = new OidcUserInfoClaimsMapper(mapperFunction);
			builder.setSharedObject(OidcUserInfoClaimsMapper.class, userInfoClaimsMapper);
		}
		return userInfoClaimsMapper;
	}
```